pytorch_lightning/utilities/distributed.py

```diff
@@ -208,6 +213,9 @@ def forward(
 
         gathered_tensor = [torch.zeros_like(tensor) for _ in range(torch.distributed.get_world_size())]
 
+        if _HPU_AVAILABLE:
+            # HPU distributed backend doesn't support int64 tensors
+            tensor = tensor.int()
```

Comment:
Should we check the tensor wasn't a float or boolean , etc.. and then apply the re-conversion. This could break user code.

Reply:
These are all indices during the initial broadcast.